### PR TITLE
Rtc wsn firmware

### PR DESCRIPTION
**Pull request by Basazababo**
Adding calibration option to fermentation tanks wsn

### CODE REVIEW

File: RTC_wireless_sensor_nodes_firmware/Fermentation_WSN/Fermentation_WSN.ino [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
-int nodeID = 21;                               // EmonTH temperature RFM12B node ID - should be unique on network
-const int networkGroup = 210;                // EmonTH RFM12B wireless network group - needs to be same as emonBase and emonGLCD
-//uint32_t networkGroup = 3421749817;    //numeric value from ansible rmc_key
+int nodeID = 23;                               // EmonTH temperature RFM12B node ID - should be unique on network
+//const int networkGroup = 210;                // EmonTH RFM12B wireless network group - needs to be same as emonBase and emonGLCD
+uint32_t networkGroup = 3421749817;    //numeric value from ansible rmc_key
                                                                       
 
-const int time_between_readings= 2;                                   // in minutes
+const int time_between_readings= 1;                                   // in minutes
```
Comment (jlark) [inline RTC_wireless_sensor_nodes_firmware/Fermentation_WSN/Fermentation_WSN.ino:20]: does this need to be changed?

File: RTC_wireless_sensor_nodes_firmware/fermentation_wsn_with_self_calibration/fermentation_wsn_with_self_calibration.ino [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+ * 11. Bring back the DIP switches to their position
+ * 12. Connect back the TX and RX of PH circuit to their previous connection
+ * 13. Connect the power supply back; the calibration is done
+ * 
+ */
+  
+#define RF69_COMPAT 1                                                              // Set to 1 if using RFM69CW or 0 is using RFM12B
+#include <JeeLib.h>                                                                      //https://github.com/jcw/jeelib - Tested with JeeLib 3/11/14
+
+boolean debug=1;                                       //Set to 1 to few debug serial output, turning debug off increases battery life
```
Comment (jlark) [inline RTC_wireless_sensor_nodes_firmware/fermentation_wsn_with_self_calibration/fermentation_wsn_with_self_calibration.ino:36]: should this be left as 1? I know it shouldn't make a big difference for the PE sensors

File: RTC_wireless_sensor_nodes_firmware/fermentation_wsn_with_self_calibration/fermentation_wsn_with_self_calibration.ino [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+        {
+            previousCalibration = current_time;
+            highpoint_cal(); 
+        }
+     }
+} 
+
+void midpoint_cal()
+{                        //calibrate to a pH of 7
+  for (i=0; i<20; i++)
```
Comment (jlark) [inline RTC_wireless_sensor_nodes_firmware/fermentation_wsn_with_self_calibration/fermentation_wsn_with_self_calibration.ino]: Perhaps we make this a varialbe as I see you're smoothing over 20 readings?

File: RTC_wireless_sensor_nodes_firmware/fermentation_wsn_with_self_calibration/fermentation_wsn_with_self_calibration.ino [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+        unsigned long current_time = millis()/60000;
+        if(current_time - previousCalibration >= calibrate_after) 
+        {
+            previousCalibration = current_time;
+            highpoint_cal(); 
+        }
+     }
+} 
+
+void midpoint_cal()
```
Comment (jlark) [inline RTC_wireless_sensor_nodes_firmware/fermentation_wsn_with_self_calibration/fermentation_wsn_with_self_calibration.ino]: You can have only one method to do all calculations. Just have a function that does

`ph_calibrate(String level)
`

And simply call it as such
`ph_calibrate('low');`

**Review by jlark — COMMENTED** [review]
ok just some minor fixes. Please test before updating.

File: RTC_wireless_sensor_nodes_firmware/Fermentation_WSN/Fermentation_WSN.ino (hunk 1 above)
Comment (Basazababo) [reply RTC_wireless_sensor_nodes_firmware/Fermentation_WSN/Fermentation_WSN.ino:20]: No need to change that,
Not sure how i changed that, am going to keep 1 min interval

File: RTC_wireless_sensor_nodes_firmware/fermentation_wsn_with_self_calibration/fermentation_wsn_with_self_calibration.ino (hunk 2 above)
Comment (Basazababo) [reply RTC_wireless_sensor_nodes_firmware/fermentation_wsn_with_self_calibration/fermentation_wsn_with_self_calibration.ino:36]: Yeah, I see, it affect the PH.
I keep it as 1

File: RTC_wireless_sensor_nodes_firmware/fermentation_wsn_with_self_calibration/fermentation_wsn_with_self_calibration.ino (hunk 4 above)
Comment (Basazababo) [reply RTC_wireless_sensor_nodes_firmware/fermentation_wsn_with_self_calibration/fermentation_wsn_with_self_calibration.ino]: I don`t understand how i can do that.
you guide me on this maybe

**Comment by jlark** [comment]
Looks good!

**Pull request merged**